core/chains/legacyevm/chain.go

```diff
@@ -244,10 +244,24 @@ func newChain(ctx context.Context, cfg *evmconfig.ChainScoped, nodes []*toml.Nod
 		}
 	}
 
-	// note: gas estimator is started as a part of the txm
-	txm, gasEstimator, err := newEvmTxm(opts.DS, cfg.EVM(), opts.AppConfig.EVMRPCEnabled(), opts.AppConfig.Database(), opts.AppConfig.Database().Listener(), client, l, logPoller, opts, headTracker, clientsByChainID)
+	// initialize gas estimator
+	gasEstimator, err := newGasEstimator(cfg.EVM(), client, l, opts, clientsByChainID)
```

Comment:
We always initialize a gas estimator even if its never going to be used?

Reply:
As far as I can tell it isn't used as of now but we don't have a NullGasEstimator to use instead. It didn't seem like an issue to just initialize an actual instance of it though since it wouldn't be started if the TXM was disabled. But at least initializing it would prevent panic if anyone were to refer to it with the TXM disabled. They'd just get a gas estimator not started error instead.